src/game/logic/system/partitionmanager.h

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file PartitionManager
+ *
+ * @author Duncans_Pumpkin
+ *
+ * @brief Stores information on current generals rank and purchased "sciences".
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "coord.h"
+#include "gametype.h"
+#include "mempoolobj.h"
+#include "snapshot.h"
+#include "subsysteminterface.h"
+#include <queue>
+
+struct FindPositionOptions;
+struct ShroudStatusStoreRestore;
+class Object;
+class GhostObject;
+class PartitionFilter;
+class SimpleObjectIterator;
+class GeometryInfo;
+class CellAndObjectIntersection;
+
+enum DistanceCalculationType : int32_t
+{
+    FROM_CENTER_2D,
+    FROM_CENTER_3D,
+    FROM_BOUNDINGSPHERE_2D,
+    FROM_BOUNDINGSPHERE_3D,
+};
+
+enum IterOrderType : int32_t
+{
+    ITER_FASTEST,
+    ITER_SORTED_NEAR_TO_FAR,
+    ITER_SORTED_FAR_TO_NEAR,
+    ITER_SORTED_CHEAP_TO_EXPENSIVE,
+    ITER_SORTED_EXPENSIVE_TO_CHEAP,
+};
+
+enum ValueOrThreat : int32_t
+{
+    VOT_THREAT,
+    VOT_VALUE,
+};
+
+enum ShroudedStatusType : int32_t
+{
+    SHROUDED_INVALID,
+    SHROUDED_NONE,
+    SHROUDED_PARTIAL,
+    SHROUDED_ALL,
+    SHROUDED_UNK4,
+};
+
+class PartitionCell : public SnapShot
+{
+public:
+    ~PartitionCell();
+
+    virtual void CRC_Snapshot(Xfer *xfer) override;
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override{};
+
+private:
+    CellAndObjectIntersection *m_firstCoilInCell;
+    uint32_t unk1[16]; // some sort of 2 16bit structure
```

Comment:
m_shroudStatus fetched by getShroudStatusForPlayer

Reply:
what is the convention for an unknown structure? Its definitily two uint16_t's unsure if its a struct or an array

Reply:
I've gone for introducing a new struct called `ShrouderLooker` there is clearly a function that is add/remove looker and and/remove shrouder (possibly shroudGeneration)